code/datums/martial/SNAP_INTO_A_SLIM_JIM.dm

```diff
@@ -0,0 +1,422 @@
+
+/datum/martial_art/wrestling/teach(var/mob/living/carbon/human/H,var/make_temporary=0)
+	..()
+	H << "<span class = 'userdanger'>SNAP INTO A SLIM JIM!</span>"
+	H << "<span class = 'danger'>Place your cursor over a move at the top of the screen to see what it does.</span>"
+	drop.Grant(H)
+	kick.Grant(H)
+	slam.Grant(H)
+	throw_wrassle.Grant(H)
+	strike.Grant(H)
+
+/datum/martial_art/wrestling/remove(var/mob/living/carbon/human/H)
+	..()
+	H << "<span class = 'userdanger'>You no longer feel that the tower of power is too sweet to be sour...</span>"
+	drop.Remove(H)
+	kick.Remove(H)
+	slam.Remove(H)
+	throw_wrassle.Remove(H)
+	strike.Remove(H)
+
+/datum/martial_art/wrestling/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(check_streak(A,D))
+		return 1
+	..()
+
+/datum/martial_art/wrestling/proc/throw_wrassle(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!D)
+		return
+	if(!A.pulling || A.pulling != D)
+		A << "You need to have [D] in a cinch!"
+		return
+	D.forceMove(A.loc)
+	D.dir = get_dir(D, A)
+
+	D.Stun(4)
+	A.visible_message("<span class = 'danger'><B>[A] starts spinning around with [D]!</B></span>")
+	A.emote("scream")
+
+	for (var/i = 0, i < 20, i++)
+		var/delay = 5
+		switch (i)
+			if (17 to INFINITY)
+				delay = 0.25
+			if (14 to 16)
+				delay = 0.5
+			if (9 to 13)
+				delay = 1
+			if (5 to 8)
+				delay = 2
+			if (0 to 4)
+				delay = 3
+
+		if (A && D)
+
+			if (get_dist(A, D) > 1)
+				A << "[D] is too far away!"
+				return 0
+
+			if (!isturf(A.loc) || !isturf(D.loc))
+				A << "You can't throw [D] from here!"
+				return 0
+
+			A.dir = turn(A.dir, 90)
+			var/turf/T = get_step(A, A.dir)
+			var/turf/S = D.loc
+			if ((S && isturf(S) && S.Exit(D)) && (T && isturf(T) && T.Enter(A)))
+				D.forceMove(T)
+				D.dir = get_dir(D, A)
+		else
+			return 0
+
+		sleep(delay)
+
+	if (A && D)
+		// These are necessary because of the sleep call.
+
+		if (get_dist(A, D) > 1)
+			A << "[D] is too far away!"
+			return 0
+
+		if (!isturf(A.loc) || !isturf(D.loc))
+			A << "You can't throw [D] from here!"
+			return 0
+
+		D.forceMove(A.loc) // Maybe this will help with the wallthrowing bug.
+
+		A.visible_message("<span class = 'danger'><B>[A] throws [D]!</B></span>")
+
+		var/turf/T = get_edge_target_turf(A, A.dir)
+		if (T && isturf(T))
+			if (!D.stat)
+				D.emote("scream")
+			D.throw_at(T, 10, 4)
+			D.Weaken(2)
+
+	return 0
+
+/datum/martial_art/wrestling/proc/slam(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(!D)
+		return
+	if(!A.pulling || A.pulling != D)
+		A << "You need to have [D] in a cinch!"
+		return
+	D.forceMove(A.loc)
+	A.dir = get_dir(A, D)
+	D.dir = get_dir(D, A)
+
+	A.visible_message("<span class = 'danger'><B>[A] lifts [D] up!</B></span>")
+
+	spawn (0)
+		if (D)
+			animate(D, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
+		sleep (15)
+		if (D)
+			animate(transform = null, time = 1, loop = 0)
+
+	for (var/i = 0, i < 3, i++)
```

Comment:
can you generalise this what I assume is rotation/spinning shit into some helper procs please

this wall of nested code makes me want to gouge out my eyes